templates/ts-apollo-fullstack/server/src/schema/schema.graphql

```diff
@@ -10,20 +10,27 @@ type Comment {
   note: Note
 }
 
-input CommentInput {
+input CommentData {
+  id: ID
+  text: String
```

Comment:
Missing mapping to required fields

Reply:
The data model has not changed and this is the correct mapping - text and description are not required in the model. ID is not required to allow both update and create using the input. To fix this we would have to create a separate input type for each when advanced CRUD updates are available with `updateComment(filter, data)`

Reply:
So then what is the point of having this change merged? We have two exactly the same input types?

Reply:
Requires discussion

Reply:
No - this is a bad example because `Comment` does not have any non-ID required fields, so the data input type does not have required fields.

You can see that `NoteData` maps required fields from `Note` because it has a required field  - Comment does not.

https://github.com/aerogear/graphback/blob/52f591e054281477c9fbd623a706690d134b2aee/templates/ts-apollo-fullstack/server/src/schema/schema.graphql#L48

With our current limited CRUD capabilities the only way to resolve this is to have a separate input type for update and delete.

- CreateCommentData - maps input fields EXCEPT for primary key
- UpdateCommentData - maps input fields WITH primary key.

We could do this now, but the work would become obsolete when we change the CRUD implementation/spec to:

`updateComment(where, data)`

Reply:
So what do you think? Create 3 input types for now and remove one at a later stage? I suppose this makes the most sense as it fixes the current implementation.

Reply:
1) Apply my suggestions and merge this knowing that this is still not getting us in the place we want to be.

2) Work on top of this issue to figure out our desired CRUD schema (possible options we can take) and propose it in demo/doc

3) Alpaca farming